source/Patches/AttackStackSequence_OnAttackComplete.cs

```diff
@@ -0,0 +1,134 @@
+
+namespace CustomSalvage
+{
+    [HarmonyPatch(typeof(AttackStackSequence), "OnAttackComplete")]
+    public static class AttackStackSequence_OnAttackComplete
+    {
+        public static bool Prepare() => Control.Instance.Settings.ShowSalvageFloaties;
+
+        [HarmonyPriority(Priority.Last)]
+        public static void Prefix(AttackStackSequence __instance, MessageCenterMessage message)
+        {
+            List<AttackDirector.AttackSequence> directorSequences = __instance.directorSequences;
+            if (directorSequences != null)
+            {
+                try
+                {
+                    AttackDirector.AttackSequence attackSequence = directorSequences[0];
+                    if (attackSequence == null) return;
+
+                    if (attackSequence?.chosenTarget is Mech mechTarget)
+                    {
+                        Helper.ProcessMech(mechTarget);
+                    }
+
+                    if (attackSequence.attacker is Mech attackerMech)
+                    {
+                        Helper.ProcessMech(attackerMech);
+                    }
+                    
+                }
+                catch (Exception ex)
+                {
+                    Log.Main.Error?.Log($"BattlefieldCSMessage.Patches.AttackStackSequence_OnAttackComplete_Patch Prefix() {ex}");
+                }
+            }
+        }
+
+        public class Helper
+        {
+            public static void ProcessMech(Mech mech)
+            {
+                if (Helper.IsDead(mech) && Helper.CanSalvage(mech))
+                {
+                    int num = Helper.SalvageParts(mech);
+                    string text;
+                    if (mech.MechDef.MechTags.Contains(Control.Instance.Settings.NoSalvageMechTag) || mech.MechDef.MechTags.Contains(Control.Instance.Settings.NoSalvageVehicleTag))
+                    {
+                        text = "NOT SALVAGEABLE";
+                    }
+                    else
+                    {
+                        text = num is > 1 or 0 ? $"{num} SALVAGEABLE PARTS" : $"{num} SALVAGEABLE PART";
+                    }
+                    mech.Combat.MessageCenter.PublishMessage(new AddSequenceToStackMessage(new ShowActorInfoSequence(mech, text, FloatieMessage.MessageNature.Inspiration, true)));
+                }
+            }
+
+            public static bool IsDead(AbstractActor actor)
+            {
+                return actor != null && (actor.IsDead || actor.IsFlaggedForDeath || actor.HasHandledDeath);
+            }
+
+            public static bool CanSalvage(AbstractActor actor)
+            {
+                return actor != null && actor.Combat.HostilityMatrix.IsEnemy(actor.Combat.LocalPlayerTeam, actor.team);
+            }
+
+            public static int SalvageParts(Mech mech)
+            {
+                int num;
+                if (mech == null)
+                {
+                    num = 0;
+                }
+                else
```

Comment:
return 0 directly if mech == null , avoids the else/converts its to a guard/early return, removes the need for num local variable declaration separate from assignment